tests/samplers_tests/tpe_tests/test_sampler.py

```diff
@@ -767,113 +768,130 @@ def test_constrained_sample_independent_zero_startup() -> None:
     sampler.sample_independent(study, trial, "param-a", dist)
 
 
-@pytest.mark.parametrize("direction", ["minimize", "maximize"])
-@pytest.mark.parametrize("constraints_enabled", [False, True])
-def test_get_observation_pairs(direction: str, constraints_enabled: bool) -> None:
+@pytest.mark.parametrize("direction, sign", [("minimize", 1), ("maximize", -1)])
```

Comment:
Do we need the parametrization for `sign`? I think it is directly calculated in the test.

Reply:
In the original code, there was a conditional branch by `direction` inside the test function (https://github.com/optuna/optuna/pull/3689/files#diff-b71dcddb8db142eb739b8f72a42c3d5784df2e8a7ca6608cb31adebe14ee62f0L799).
I thought eliminating it is better because conditional branch by parametrized values in tests reduces readability and is not a good practice in general.

Reply:
Thanks. Could you remove it in this PR?

Reply:
Oh, I see. IMO, since this is a one line conditional branch, we can accept it. If anything, I think the previous version is simpler and sufficiently readable. @toshihikoyanase What do you think?

Reply:
> IMO, since this is a one line conditional branch, we can accept it.

I agree with @HideakiImamura . In general, we try to remove conditional branches from test cases, but the following line is simple enough, and managing the pairs of direction and sign by hand might increase the maintenance cost.

```python
sign = 1 if direction == "minimize" else -1
```

Reply:
Then, I followed your comments! PTAL.